src/node_crypto.cc

```diff
@@ -3797,9 +3797,7 @@ void Hmac::HmacDigest(const FunctionCallbackInfo<Value>& args) {
 
   enum encoding encoding = BUFFER;
   if (args.Length() >= 1) {
-    encoding = ParseEncoding(env->isolate(),
-                             args[0]->ToString(env->isolate()),
-                             BUFFER);
+    encoding = ParseEncoding(env->isolate(), args[0], BUFFER);
```

Comment:
@tniessen Still, if we drop `toString()` support it’s easy to just drop the checks again. (Likewise, it would be good if you included the test you said you had prepared in this PR – it’s easy to change things later, but for now it’s best to test the current behaviour.)

> The above only applies to `hmac` and `hash`. For `sign` and `verify`, the encoding parameter seems to always be `null`, so the `CHECK` would fail, right?

It looks that way, yes… maybe you could fix that up and drop the unused code bits? That also simplifies `SignFinal()`, right now the `StringBytes::Encode()` encode call there seems like a completely unnecessary extra copy operation.